lib/charms/archive_auth_mirror/reprepro.py

```diff
@@ -0,0 +1,40 @@
+import getpass
+
+from charmhelpers.core.templating import render
+
+from .utils import get_paths
+
+
+def configure_reprepro(mirror_uri, mirror_archs, mirror_key_fingerprint,
+                       sign_key_fingerprint, get_paths=get_paths):
+    """Create reprepro configuration files."""
+    paths = get_paths()
+    context = split_repository_uri(mirror_uri)
+    context.update(
+        {'archs': mirror_archs,
+         'mirror_key': mirror_key_fingerprint,
+         'sign_key': sign_key_fingerprint})
+    # explicitly pass owner and group for tests, otherwise root would be used
+    owner = group = getpass.getuser()
```

Comment:
yeah it's not ideal but it's normally the case (and in real use the code runs as root/root)